File: assets/cs/ws/up-client/templates/services/WsApi.cs
using nex.ws;

namespace {{namespace}}
{
    public class WSApi<TUser, TToken> : WSApiBase<TUser, TToken>
    {
        #region [ private ]
        {{#services}}
        private {{serviceUpperName}}WSService<TUser, TToken> _{{serviceName}};
        {{/services}}
        #endregion

        #region [ services ]
        {{#services}}
        public {{serviceUpperName}}WSService<TUser, TToken> {{serviceName}} { get { return _{{serviceName}}; } }
        {{/services}}
        #endregion

        #region [ constructor ]
        public WSApi()
        {
            {{#services}}
            _{{serviceName}} = new {{serviceUpperName}}WSService<TUser, TToken>(Rest, Hub);
            {{/services}}
        }
        #endregion
    }
}

File: assets/ts/ws/up-client/templates/services/wsapi.ts
import { lib } from '../'

{{#services}}
import { {{serviceUpperName}}WSService } from './services/{{serviceUpperName}}WSService'
{{/services}}

export class WSApi<TUser, Token> extends lib.WSApiBase<TUser, Token> {
    {{#services}}
    public readonly {{serviceName}} = new {{serviceUpperName}}WSService ( this.rest, this.hub )
    {{/services}}

    constructor ( ws: lib.IWSBase ){
        super( ws )
    }
}
